### ISSUE TITLE
Could not open cherrytree.exe.log

### ISSUE THREAD

**Issue by stanleyrhodes** [giuspen/cherrytree#966]
This issue started with the exact same issue reported here:
https://github.com/giuspen/cherrytree/issues/416

I would try to open a cherrytree file, it would tell me it couldn't open the cherrtree.exe.log in C:\Program Files (x86)\CherryTree\bin but I couldn't find that log file there. That seems like a bug or some sort of issue.

Then I followed the guidance in the above post to "Run as administrator" cherrytree.exe and a log appeared.

The log is as follows:
Traceback (most recent call last):
  File "cherrytree", line 78, in <module>
  File "cherrytree", line 50, in f_main
  File "main.pyc", line 248, in main
  File "main.pyc", line 83, in __init__
  File "main.pyc", line 88, in window_open_new
  File "core.pyc", line 252, in __init__
  File "core.pyc", line 1500, in file_startup_load
  File "core.pyc", line 2002, in file_load
  File "ctdb.pyc", line 650, in read_db_full
  File "ctdb.pyc", line 563, in get_node_row_partial_from_id
sqlite3.DatabaseError: database disk image is malformed

The file that caused it, I believe, was originally an XML .ctd file, which after it grew quite large, I saved as a .ctb. I've been using it since that conversion, about 3 weeks ago, without problem. I'm a bit nervous because this is a very important file for my life and daily work.

**Comment by giuspen** [comment]
Can you try to run the testing version 0.99.2 which has additional checks on the database and report the result? Look at https://www.giuspen.com/cherrytree/#downl and scroll down.
Additionally you should have the backups in the same directory to try and open.

**Comment by stanleyrhodes** [comment]
I downloaded the 0.99.2_win64_portable version, unzipped, ran as administrator, and opened the problem .ctb file. I got these messages:
"The database file P:/Documents/stan_personal_notebook.ctb is corrupt, see log for more details"
"Error parsing the cherrytree file."

I looked around for a new log file, but I can't find one. Does the portable version put it somewhere not-obvious, or did I need to do something differently?

**Comment by txe** [comment]
Hi,
The cherrytree file itself is a sqlite database, and it's corrupted. You can open it by [sqlitebrowser](https://sqlitebrowser.org/).
[This article ](https://techblog.dorogin.com/sqliteexception-database-disk-image-is-malformed-77e59d547c50) describes how to recovery the db.

**Comment by stanleyrhodes** [comment]
I am a SQL novice, so it's not obvious how to glue together the information you gave me into an entire process to get a recovered, working .ctb file. I was able to open the corrupted .ctb it with sqlitebrowser but I'm not sure what I was supposed to do with that.

I used the article you linked to download sqlite3.exe and I followed the instructions on that page. Using the integrity check it told me the following:
`*** in database main ***`
`On tree page 1156 cell 0: 2nd reference to page 1325`
`Page 1070 is never used`
[all the pages between the above and below page numbers here]
`Page 1155 is never used`

Then I followed the rest of the instructions to grab all the data and put it into a new file, called, in accordance to the instructions `dump_all.sql`. Seemingly this contains all my data in a series of INSERTs. However, I cannot find a way to import that / executre that sql into cherrytree file. I also attempted things like creating a copy of the broken .ctb as fixed_copy.ctb and then executing in the cmd line `sqlite3.exe fixed_copy.ctb ".read dump_all.sql"` which gave me a long list of errors.

How do I go from this recovered sql file to a cherrytree .ctb file?

**Comment by txe** [comment]
When you removed `BEGIN TRANSACTION;` there were still errors in reading `dump_all.sql` but they were skipped.

I suppose the issue were caused by syncing, xml is definitely safer to use with cloud services.  

About the log file, when cherrytree.exe has issues, it tries to write information in log file which place is in the same directory as cherrytree.exe is. It works OK for portable versions, but not for versions installed in `C:\Program Files (x86)` because user process cannot create files (e.g. log files) there. So cherrytree.exe cannot handle it and stop working. There is two ways to fix it:
- you can change privileges for `C:\Program Files (x86)\CherryTree\bin`, so user process can create and write files there.
- cherrytree.exe should create log files in other place.

**Comment by stanleyrhodes** [comment]
Thank you. Good to know that XML might be a better choice for cloud storage. I had wrongly assumed the two formats would be equivalently vulnerable to problems on cloud services. I much prefer the speed of sqlite but not if I'm going to lose work every month or two from a hiccup.

I think more novice users are likely to install CherryTree in the default way, which means the default functionality right now is failing to create a log on Windows. CherryTree already creates `C:\Users\[Username]\AppData\Local\cherrytree` which contains `config.cfg`. Perhaps this is the ideal place to create the log files. But I know very little about permissions on Windows.

**Comment by giuspen** [comment]
that log is actually automatically generated from the old library, the new version won't have this issue. If you want to try it you can find it already even if still under testing at https://www.giuspen.com/cherrytree/#downl scroll down see 0.99.3

**Comment by txe** [comment]
I suppose, the topic is concluded, so closing

**Comment by stanleyrhodes** [comment]
I've had this happen twice more since I posted this issue, including today. I'm still using this ctb file with a cloud service so I guess I'm rolling the dice. But I've noticed a pattern: every time it has happened, including that first time, I had pasted in in some text straight from a webpage. This time, I knew exactly what I had posted before it corrupted: the body text at https://www.europeansocialsurvey.org/about/

That page, in the third bullet point paragraph, has right single quotation marks rather than the regular typewriter apostrophe. Is there a chance there's an encoding issue when CherryTree takes pasted text with unicode characters like this and tries to save them in the sql-lite DB?
